umap/static/umap/js/umap.controls.js

```diff
@@ -966,6 +966,30 @@ L.U.AttributionControl = L.Control.Attribution.extend({
 
 });
 
+L.U.StarControl = L.Control.extend({
+
+    options: {
+        position: 'topleft'
+    },
+
+    onAdd: function (map) {
+        var status = map.options.starred ? ' starred' : '';
+        var container = L.DomUtil.create('div', 'leaflet-control-star umap-control' + status),
+            link = L.DomUtil.create('a', '', container);
+        link.href = '#';
+        link.title = L._('Star this map');
+        var fn = function () {
+            map.star()
+        };
+
+        L.DomEvent
+            .on(link, 'click', L.DomEvent.stop)
+            .on(link, 'click', fn, map)
```

Comment:
I'm not sure you need to define a `fn` here, `.on(link, 'click', map.star, map)` should work?

Reply:
Done! :)